fsw/cfe-core/src/inc/cfe_tbl_filedef.h

```diff
@@ -64,6 +64,12 @@ typedef struct
     char        TableName[CFE_MISSION_TBL_MAX_FULL_NAME_LEN]; /**< \brief Name of Table as defined onboard */
     char        Description[CFE_FS_HDR_DESC_MAX_LEN];       /**< \brief Description of table image that is included in cFE File Header */
     char        TgtFilename[CFE_MISSION_MAX_FILE_LEN];      /**< \brief Default filename to be used for output of elf2cfetbl utility  */
+
+    /** \brief  Padding to ensure that TgtFilename is null-terminated. If TgtFilename is initialized with a string of exactly
+     *          CFE_MISSION_MAX_FILE_LEN characters, the null of the initializer is lost and the string is no longer null-terminated.
+     */
+    uint32      Padding;
+
```

Comment:
I don't like that the reader is still overrunning the buffer and just happens to hit a zero "Padding" to stop it.  What about `TableName` and `Description` fields?  Wouldn't they have the same problem?

Recommend fixing the code that is actually reading this and overrunning the buffer.

Reply:
Technically it's not that the buffer is being overrun, it's that C allows you to do:

`struct foo { char a[10]; int b }; foo bar = {"0123456789", 10};`

If you do `foo bar = {"0123456789a", 10};` it throws a compiler error. So this is somewhat an edge case of having the exact wrong number of chars. At least for the other strings, having that exact wrong number results in a longer string but with the case of filename, it gave me a weird file when elf2cfetbl processed it.

The problem is that the CFE_TBL_FILEDEF is a macro that just defines a struct with a static initialization that is compiled into the symbol table. Otherwise we'd be talking about rearchitecting the table system.

I propose I add padding between all of the strings. While I could use int32/uint32, I don't see a reason to not use uint8/char?

Reply:
Yeah, its a fixed-length string buffer, there should be no assumption that it has a null terminating character on it.  No fixed length buffers (either in messages or table files or anywhere else) should be assumed to be null terminated unless the reader makes a copy and enforces it.  That's just the nature of C strings.

Adding zero padding between every field is very hack-ish and doesn't really address the problem that `elf2cfetbl` (or whatever reads this struct) isn't properly reading these fields as fixed length string buffers.  No code should be allowed to overrun the buffer it is reading.  That's just bad.

It just means `elf2cfetbl` should copy the strings, adhering to the fixed length limit, before using them or passing them through to C library calls.  This is what CFE FSW does, all tools should do the same.